lib/memory.js

```diff
@@ -58,9 +60,12 @@ module.exports.startMemoryMeasurement = function (test, options, time, graphData
 
         var memoryResult, initialResult;
 
-        var filename = 'M_' + test.description.replace(/\W/gi, '-') + '_' + time + '.txt';
-        var csvFilename = 'M_' + test.description.replace(/\W/gi, '-') + '_' + time + '.csv';
-        var graphFilename = 'M_' + test.description.replace(/\W/gi, '-') + '_' + time + '.png';
+        // Try to create the directory first
+        mkdirp.sync(options.dir);
+
+        var filename = options.dir + 'M_' + test.description.replace(/\W/gi, '-') + '_' + time + '.txt';
```

Comment:
This will create a file called `./docs/memoryM_...`, since you're missing a slash between the dir and the file name.

Please use `path.join` for this, as it will automatically put in the required directory separator.